forte/processors/data_augment/text_generation_augment_processor.py

```diff
@@ -0,0 +1,132 @@
+# Copyright 2020 The Forte Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Processors that augment the data.
+"""
+import random
+import string
+import nltk
+from nltk.tokenize.treebank import TreebankWordTokenizer, TreebankWordDetokenizer
+from typing import Iterable, Tuple, Dict
+from abc import abstractmethod, ABC
+from forte.data.data_pack import DataPack
+from forte.data.multi_pack import MultiPack
+from forte.common.resources import Resources
+from forte.common.configuration import Config
+from forte.processors.base.data_augment_processor import BaseDataAugmentProcessor
+from ft.onto.base_ontology import (
+    Token, Sentence, Document
+)
+
+
+__all__ = [
+    "TextGenerationDataAugmentProcessor"
+]
+
+random.seed(0)
+
+class TextGenerationDataAugmentProcessor(BaseDataAugmentProcessor):
+    r"""
+    The data augmentation processor for text generation data.
+    The multipack contains two datapacks: source and target.
+    This class augment both datapacks and add the new datapacks to
+    the original multipack as augmented source and target.
+    """
+    def initialize(self, resources: Resources, configs: Config):
+        super().initialize(resources, configs)
+        self.tokenizer = TreebankWordTokenizer()
+        self.detokenizer = TreebankWordDetokenizer()
```

Comment:
So here you are requiring the users to use NLTK tokenizer, this is not what we want. We expect the system would use whatever tokenization created from the previous processors, not to force the users to use NLTK here.